### PR TITLE
Frontend: refactor `Balance` component

### PR DESCRIPTION
**Pull request by BrianCraig**
- Migrate `ClassComponent` to `FunctionComponent`
 - Migrate HOCs tos hooks

### CODE REVIEW

File: frontends/web/src/components/balance/balance.tsx [hunk 1]
```diff
@@ -15,22 +15,20 @@
...
-interface BalanceProps {
+interface Props {
     balance?: IBalance;
 }
 
-type Props = BalanceProps & TranslateProps;
-
-function Balance({
-    t,
+export const Balance: FunctionComponent<Props> = ({
```
Comment (thisconnect) [inline frontends/web/src/components/balance/balance.tsx:28]: @benma do you ack to renaming 'CompomentNameProps' to just 'Props' or is it helpful if it stays, i.e. in this case `BalanceProps`?

File: frontends/web/src/components/balance/balance.tsx [hunk 2]
```diff
@@ -15,22 +15,20 @@
...
+import { useTranslation } from 'react-i18next';
 import { IBalance } from '../../api/account';
 import { FiatConversion } from '../../components/rates/rates';
-import { translate, TranslateProps } from '../../decorators/translate';
 import style from './balance.module.css';
 
-interface BalanceProps {
+interface Props {
     balance?: IBalance;
 }
```
Comment (thisconnect) [inline frontends/web/src/components/balance/balance.tsx:26]: @BrianCraig if you touch it anyway, would you mind changing to  `type Props = {` ?

**Review by thisconnect — APPROVED** [review]
LGTM with suggestion

File: frontends/web/src/components/balance/balance.tsx (hunk 1 above)
Comment (benma) [reply frontends/web/src/components/balance/balance.tsx:28]: sure, ACK.

**Pull request merged**